### PR TITLE
chore: migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by JenniferMah**
I removed code climate since we no longer have the associated credentials and the current configuration in our Travis CI builds do not generate a code coverage report.
Jira Ticket: [DI-1805](https://issues.corp.twilio.com/browse/DI-1805)

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Build & Test
+        run: make test-docker test-install
+      - run: bash <(curl -s https://codecov.io/bash)
```
Comment (eshanholtz) [inline .github/workflows/tests.yml:33]: remove this if we're not doing codecov

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Publish Python distributions
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    name: Deploy
```
Comment (eshanholtz) [inline .github/workflows/release.yml:10]: update all the 'deploy's to 'release'

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build
```
Comment (eshanholtz) [inline .github/workflows/release.yml:24]: why not `make install` here?

Comment (JenniferMah) [reply .github/workflows/release.yml:24]: We don't need the virtual environment so I opted for a `pip install build` instead of the make install.

**Review by eshanholtz — APPROVED** [review]

**Pull request merged**